clarify specifics of webhook support

gabemontero opened the pull request:
Fixes https://github.com/openshift/openshift-docs/issues/4290

@bparees @oatmealraisin ptal

bparees commented:
Does the doc somewhere explain that the build that is run as a result of the hook is going to build exactly the commit/ref provided in the hook event?

oatmealraisin commented:
@gabemontero lgtm 👍

gabemontero commented:
@bparees @oatmealraisin  I've reworked to make the build's use of the webhook event payload more explicit per the request from @bparees  (I did not find where it was explicitly noted before).

Also, moved off of the use of NOTE's.

oatmealraisin commented:
@gabemontero This also lgtm 👍

bparees commented on dev_guide/builds/triggering_builds.adoc:
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
 link:https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html[Bitbucket],
 or Generic webhooks.
 
+{product-title} webhooks currently only supports their analogous versions of the push event for each of the Git based source code management
```
s/supports/support/

bparees commented on dev_guide/builds/triggering_builds.adoc:
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
+systems (`SCMs`).  All other event types are ignored.
+
+When the push events are processed, a confirmation is made as to whether the branch reference inside the event matches the branch reference in the corresponding
+`BuildConfig`.  If they match, then the exact commit reference noted in the webhook event is checked out for the {product-title} build.
```
"If they do not match, no build is triggered."

gabemontero commented:
updates pushed

bparees commented:
technically applies to earlier versions of OCP too but since the new hooks were introduced in 3.6 it's not worth the hassle of refactoring the doc for different versions.

bparees commented on dev_guide/builds/triggering_builds.adoc:
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
+systems (`SCMs`).  All other event types are ignored.
+
+When the push events are processed, a confirmation is made as to whether the branch reference inside the event matches the branch reference in the corresponding
+`BuildConfig`.  If they do not match, no build is triggered.
```
i think you lost the part where it said we will build the specific commit ref in the event?

gabemontero replied:
I misinterpreted your prior comment then.  I'll revert and then attempt to reapply your last comment.

bparees replied:
yes, sorry it was intended as an addendum, not a replacement... sloppy commenting on my part.

gabemontero commented:
update pushed (re-do on interpreting that last comment)

bparees commented:
lgtm, @openshift/team-documentation ptal/merge

vikram-redhat commented:
@bfallonf - PTAL.

adellape commented on dev_guide/builds/triggering_builds.adoc:
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
 or Generic webhooks.
 
+{product-title} webhooks currently only support their analogous versions of the push event for each of the Git based source code management
+systems (`SCMs`).  All other event types are ignored.
```
I don't think SCMs needs to be marked up here.

gabemontero replied:
update pushed

bfallonf commented on dev_guide/builds/triggering_builds.adoc:
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
+{product-title} webhooks currently only support their analogous versions of the push event for each of the Git based source code management
+systems (SCMs).  All other event types are ignored.
+
+When the push events are processed, a confirmation is made as to whether the branch reference inside the event matches the branch reference in the corresponding
```
@gabemontero Is it OpenShift that confirms the above?

gabemontero replied:
@bfallonf - yes, the confirmation is made within openshift, specifically the openshift master.

thanks

bfallonf replied:
@gabemontero Thanks. I'll merge this and do a quick followup to include that.

bfallonf commented:
Thanks @gabemontero . I just have one question about the process then I can probably merge this.
